Sources/MongoSwift/BSON/BsonEncoder.swift

```diff
@@ -507,16 +509,24 @@ private class MutableArray: BsonValue {
 
     var count: Int { return array.count }
 
+    func insert(_ value: BsonValue?, at index: Int) {
+        self.array.insert(value, at: index)
+    }
+
     func encode(to data: UnsafeMutablePointer<bson_t>, forKey key: String) throws {
         try self.array.encode(to: data, forKey: key)
     }
 
+    /// methods required by the BsonValue protocol that we don't actually need/use. MutableArray
```

Comment:
Should these methods be annotated with `required`, or is that only used for initializers?